resources/styles/components/cc-dashboard/index.less

```diff
@@ -159,50 +149,22 @@
       margin-left: 20px;
       .banner       { fill: @openstax-book-default-bg; }
       .banner-light { fill: lighten(@openstax-book-default-bg, 10%); }
-      .course-name{
+      .course-name {
         font-family:'Helvetica bold', Lato;
         font-size: 15px;
         fill: @openstax-book-default-fg;
       }
-      &.principles_economics {
-        .banner       { fill: @principles_economics-bg; }
-        .banner-light { fill: lighten(@principles_economics-bg, 10%); }
-        .course-name  { fill: @principles_economics-fg; }
-      }
-      &.macro_economics {
-        .banner       { fill: @macro_economics-bg; }
-        .banner-light { fill: lighten(@macro_economics-bg, 10%); }
-        .course-name  { fill: @macro_economics-fg; }
-      }
-      &.college_physics {
-        .banner       { fill: @college_physics-bg; }
-        .banner-light { fill: lighten(@college_physics-bg, 10%); }
-        .course-name  { fill: @college_physics-fg; }
-      }
-      &.biology {
-        .banner       { fill: @biology-bg; }
-        .banner-light { fill: lighten(@biology-bg, 10%); }
-        .course-name  { fill: @biology-fg; }
-      }
-      &.micro_economics {
-        .banner       { fill: @micro_economics-bg; }
-        .banner-light { fill: lighten(@micro_economics-bg, 10%); }
-        .course-name  { fill: @micro_economics-fg; }
-      }
-      &.concepts_biology {
-        .banner       { fill: @concepts_biology-bg; }
-        .banner-light { fill: lighten(@concepts_biology-bg, 10%); }
-        .course-name  { fill: @concepts_biology-fg; }
-      }
-      &.intro_sociology {
-        .banner       { fill: @intro_sociology-bg; }
-        .banner-light { fill: lighten(@intro_sociology-bg, 10%); }
-        .course-name  { fill: @intro_sociology-fg; }
+
+      .apply-svg-image-styling(@code; @fg; @bgvar) {
+        @bg: @@bgvar; // lighten doesn't like double @@
+        &.@{code}{
+          .banner       { fill: @bg; }
+          .banner-light { fill: lighten(@bg, 10%); }
+          .course-name  { fill: @@fg; }
+        }
       }
-      &.anatomy_physiology {
-        .banner       { fill: @anatomy_physiology-bg; }
-        .banner-light { fill: lighten(@anatomy_physiology-bg, 10%); }
-        .course-name  { fill: @anatomy_physiology-fg; }
+      .for(@openstax-book-appearance-codes); .-each(@code) {
```

Comment:
Yeah, I ran across the `.for`mixin and think it expresses what's going on better.  Internally it's using that pattern of course,  as far as I know that's the only real way to loop in less.